external/common/share/capnproto/README.md

```diff
@@ -1,2 +1,6 @@
-capnproto-c++-0.5.3.tar.gz: from https://capnproto.org/capnproto-c++-0.5.3.tar.gz
-capnproto-c++-win32-0.5.3.zip: from https://capnproto.org/capnproto-c++-win32-0.5.3.zip
+capnproto-c++-0.5.3.tar.gz: from https://capnproto.org/capnproto-c++-0.5.3.1.tar.gz
+capnproto-c++-win32-0.5.3.zip: from https://capnproto.org/capnproto-c++-win32-0.5.3.1.zip
+
+When pycapnp (in requirements.txt) is updated, please check with which version of capnproto it ships and update the files also here
+and in external/Capnproto.cmake ! 
```

Comment:
> clang 4.x breaks when these two capnp versions are not in sync, and it isn't documented in nupic.core readmes

@breznak, you should be able to build nupic.core without having pycapnp on your system at all. The nupic.core master branch build should only depend on headers and from capnproto that is bundled inside nupic.core (https://github.com/numenta/nupic.core/tree/master/external/common/share/capnproto) and link with archives produced from the same.

Could you please look into why the cmake build breaks with clang 4.x? Perhaps there is some unintentional reference to pycapnp (headers?) accidentally left in nupic.core's cmake configuration that should be removed.

Also, we cannot depend on the capnp .so that is shipped with pycapnp for nupic.core builds, because we have no control over which pycapnp (and by extension capnp .so) the user has on their system. The APIs might be incompatible, etc. Also, we cannot use the .so, because we run into duplicate schema ID issue related to how capnproto registers templated schema representations of same schema that is built/used by more than one extension.